numerical_methods/secant_method.cpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * \file
+ * \brief Solve the equation \f$f(x)=0\f$ using
+ * [Secant Method](https://en.wikipedia.org/wiki/Secant_method)
+ *
+ * The example function \f$f(x)\f$ looks like this:
+ * \f[
+ * x^4 - 29 = 0
+ * \f]
+ *
+ * \author [Fatih Küçükkarakurt](https://github.com/fkkarakurt)
+ */
+
+// Secant Method for the root of f(x)
+// Analytically the solution is x^4-29 = 0
+// => x = 29^(1/4) = 2,3205957871060836757951626724362...
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+
+double f(double x) { return x * x * x * x - 29.0; } // The Function f(x)
```

Comment:
Please add documentation with detailed/brief description, parameters description, etc. etc..
Check our [contributing guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md). You can see an example of the same here:
```c++
/**
 * Utility function to print matrix
 * @tparam n number of matrix size
 * @param board matrix where numbers are saved
 */
template <size_t n>
void printSolution(const std::array<std::array<int, n>, n> &board) {
    std::cout << "\n";
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            std::cout << "" << board[i][j] << " ";
        }
        std::cout << "\n";
    }
}
```